Sources/ToolchainRegistry/ToolchainRegistry.swift

```diff
@@ -283,7 +286,18 @@ package final actor ToolchainRegistry {
   /// If we have a toolchain in the toolchain registry that contains the compiler with the given URL, return it.
   /// Otherwise, return `nil`.
   package func toolchain(withCompiler compiler: URL) -> Toolchain? {
-    return toolchainsByCompiler[compiler]
+    if let resolvedPath = try? compiler.realpath {
+      return toolchainsByCompiler[resolvedPath]
```

Comment:
We should run `realpath` when populating `toolchainsByCompiler` as well to cover symlinks within the toolchain, eg. `swift` and `swiftc` to `swift-driver`.

Reply:
Ah, good point. I didn't do this since we already realpathed the toolchain path coming into ToolchainRegistry, but forgot about the case the binaries themselves could be. We could just realpath the parent folder here?

Reply:
I think it’s fair enough to assume that `bin` is not a symlink. If somebody starts symlinking `bin`, I really want to know what they are doing.